Add Multi-agent demo

finani opened the pull request:
Add Multi-agent demo which has 6 jackals on warehouse map.
- All objects including jackal and map is from mvsim.
- File tree also follows the one of mvsim.
- Use fake-amcl (ground truth on mvsim)
- vehicle name follows default vehicle name (veh1, veh2, ...)
- Namespace of nav2 is vehicle name. (namespace:=veh1)

finani committed:
copy demo_warehouse from mvsim (+rename)

finani committed:
add 6 robots demo

finani committed:
add objects and follow mvsim folder tree

finani committed:
fix path, update readme

finani committed:
clean up vehs params

jlblancoc commented:
Awesome!! 

Just a couple first comments: 
1) It runs too slow for my PC :-) I guess your GPU is more modern, but mine struggled to get going with MVSim and hit ~110% of CPU usage. Would you consider reducing the number of 3D LiDAR rays to 16 per robot? or, if they are not being used, reduce a bit the number of cameras / 3D sensors? Until you see a reduction in the CPU usage of ~10-20% in your machine? Just to make things smoother to other people with slower GPUs like me :-) 

2) Running this on iron I got:

    [component_container_isolated-1] [ERROR] [1726043446.657506886] [veh1.bt_navigator]: Exception when loading BT: Error at line 14: -> Node not recognized: WouldAPlannerRecoveryHelp

I guess this is intended to work on Nav2 version on Humble?? If so, please add it to the README. 


3) Although it's standard for Nav2, would you add a couple of lines about how a user might interact with the RViz interface to launch Nav2's navigation?

PS: After those changes I'll keep testing... again, thanks for this great contribution :100:

finani committed:
remove camera and 3d lidar

finani committed:
update readme (add how to send a goal)

finani committed:
remove explicitly defined bt navigator plugins

finani committed:
update readme (humble only)

finani commented:
Thank you for the detailed comments!
I tried to fix everything, but there were a few things I couldn't resolve.

1. I removed camera and 3d lidar not necessary by nav2. I tested at i7-9750H with RTX2080 Max-Q, but It looks good.

2. I removed explicitly defined plugin names to solve this issue, but I got [another issue](https://github.com/ros2/rviz/issues/463). In my labtop, this is also occured in Jazzy. So, I wrote about this in readme.

```bash
[rviz2-2] [ERROR] [1726885343.929317692] [rviz2]: rviz/glsl120/indexed_8bit_image.vert
[rviz2-2] rviz/glsl120/indexed_8bit_image.frag
[rviz2-2]  GLSL link result : 
[rviz2-2] active samplers with a different type refer to the same texture image unit
```

3. I wrote about how to send a goal using rviz and cli including goal poses action for waypoint following.

jlblancoc commented:
@finani Thanks a lot for this fantastic contribution! I thought there was something missing but just tested it and worked like a charm :fire:
